Test tweaks

erp12 opened the pull request:
Resolves issues 2 through 4 of #210.

Happy to make any adjustments that you would like!

erp12 committed:
Changes date-time tests to be agnostic to timezone

erp12 committed:
Changes temp directory creation to java.nio utility

erp12 committed:
Relaxes JavaSparkContext spark-home assertion in tests

anthony-khong commented on test/zero_one/geni/rdd_test.clj:
```diff
@@ -53,7 +53,7 @@
     (rdd/persistent-rdds) => map?
     (rdd/resources) => {}
-    (rdd/spark-home) => nil?
+    (rdd/spark-home) => any?
```
Actually, I believe this should be checked against the real Spark home using `(System/getenv "SPARK_HOME")`. Apologies, I should've said on the original issue thread!

erp12 replied:
Good point, that is much better. Thanks! Just fixed it.

anthony-khong reviewed (commented):
Thank you so much for this! This is awesome! I just have one tiny comment on `any?`. I hope that's fine!

anthony-khong commented:
Hi Eddie, I believe the next time you run the CI pipeline, the `lint-ancient` ancient job will fail. You will have to upgrade the `lein-cloverage` to `1.2.1` in `project.clj`.

erp12 commented:
@anthony-khong Thanks for the review. I have addressed the PR comment and rebased on master so that the `lein-cloverage` version bump is in the history. Let me know if there is anything else you would like me to do.

anthony-khong commented:
Thanks again @erp12. This is awesome!
